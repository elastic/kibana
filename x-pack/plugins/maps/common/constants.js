/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export const GIS_API_PATH = 'api/maps';
export const EMS_DATA_FILE_PATH = 'ems/file';
export const EMS_DATA_TMS_PATH = 'ems/tms';
export const EMS_META_PATH = 'ems/meta';

export const MAP_SAVED_OBJECT_TYPE = 'map';
export const APP_ID = 'maps';
export const APP_ICON = 'gisApp';

export function createMapPath(id) {
  return `/app/maps#/map/${id}`;
}

export const EMS_FILE = 'EMS_FILE';
export const ES_GEO_GRID = 'ES_GEO_GRID';
export const ES_SEARCH = 'ES_SEARCH';
export const SOURCE_DATA_ID_ORIGIN = 'source';

export const DECIMAL_DEGREES_PRECISION = 5; // meters precision
export const ZOOM_PRECISION = 2;
export const DEFAULT_ES_DOC_LIMIT = 2048;

export const FEATURE_ID_PROPERTY_NAME = '__kbn__feature_id__';

export const ES_GEO_FIELD_TYPE = {
  GEO_POINT: 'geo_point',
  GEO_SHAPE: 'geo_shape'
};

export const GEO_JSON_TYPE = {
  POINT: 'Point',
  MULTI_POINT: 'MultiPoint',
  LINE_STRING: 'LineString',
  MULTI_LINE_STRING: 'MultiLineString',
  POLYGON: 'Polygon',
  MULTI_POLYGON: 'MultiPolygon',
  GEOMETRY_COLLECTION: 'GeometryCollection',
};
