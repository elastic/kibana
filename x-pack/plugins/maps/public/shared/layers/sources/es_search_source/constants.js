/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export const DEFAULT_ES_DOC_LIMIT = 2048;

export const DEFAULT_FILTER_BY_MAP_BOUNDS = true;
