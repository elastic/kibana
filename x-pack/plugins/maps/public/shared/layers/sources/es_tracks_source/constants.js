/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { i18n } from '@kbn/i18n';

export const indexPatternLabel = i18n.translate('xpack.maps.source.esTracks.indexPatternLabel', {
  defaultMessage: `Index pattern`,
});

export const geoFieldLabel = i18n.translate('xpack.maps.source.esTracks.geoFieldLabel', {
  defaultMessage: 'Point field',
});

export const timeFieldLabel = i18n.translate('xpack.maps.source.esTracks.timeFieldLabel', {
  defaultMessage: 'Time field',
});

export const splitFieldLabel = i18n.translate('xpack.maps.source.esTracks.splitLabel', {
  defaultMessage: 'Split field',
});
