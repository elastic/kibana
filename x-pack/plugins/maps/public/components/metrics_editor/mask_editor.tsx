/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import React, { Component } from 'react';
import { i18n } from '@kbn/i18n';
import { EuiExpression, EuiPopover } from '@elastic/eui';
import { AGG_TYPE } from '../../../common/constants';
import { AggDescriptor } from '../../../common/descriptor_types';

interface Props {
  metric: AggDescriptor;
  onChange: (metric: AggDescriptor) => void;
}

interface State {
  isPopoverOpen: boolean;
}

export class MaskEditor extends Component<Props, State> {
  state: State = { isPopoverOpen: false };

  _togglePopover = () => {
    this.setState((prevState) => ({
      isPopoverOpen: !prevState.isPopoverOpen,
    }));
  };

  _closePopover = () => {
    this.setState({
      isPopoverOpen: false,
    });
  };

  render() {
    // masks only supported for numerical metrics
    if (this.props.metric.aggType === AGG_TYPE.TERMS) {
      return null;
    }

    const expressionValue =
      this.props.metric.mask !== undefined
        ? 'less than 2'
        : '--';

    return (
      <EuiPopover
        id="mask"
        button={
          <EuiExpression
            color={this.props.metric.mask === undefined ? 'subdued' : 'warning'}
            description={i18n.translate('xpack.maps.maskEditor.maskDescription', {
              defaultMessage: 'hide {bucketsLabel} when {aggLabel} is ',
              values: {
                bucketsLabel: 'clusters',
                aggLabel: 'count',
              },
            })}
            value={expressionValue}
            onClick={this._togglePopover}
            uppercase={false}
          />
        }
        isOpen={this.state.isPopoverOpen}
        closePopover={this._closePopover}
        panelPaddingSize="s"
        anchorPosition="downCenter"
        repositionOnScroll={true}
      >
        operator select and value input go here
      </EuiPopover>
    );
  }
}
