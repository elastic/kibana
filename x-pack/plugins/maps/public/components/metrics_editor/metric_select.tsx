/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import React from 'react';
import { i18n } from '@kbn/i18n';
import { EuiComboBox, EuiComboBoxOptionOption, EuiComboBoxProps } from '@elastic/eui';
import { AGG_TYPE } from '../../../common/constants';

const AGG_OPTIONS = [
  {
    label: i18n.translate('xpack.maps.metricSelect.averageDropDownOptionLabel', {
      defaultMessage: 'Average',
    }),
    value: AGG_TYPE.AVG,
  },
  {
    label: i18n.translate('xpack.maps.metricSelect.countDropDownOptionLabel', {
      defaultMessage: 'Count',
    }),
    value: AGG_TYPE.COUNT,
  },
  {
    label: i18n.translate('xpack.maps.metricSelect.maxDropDownOptionLabel', {
      defaultMessage: 'Max',
    }),
    value: AGG_TYPE.MAX,
  },
  {
    label: i18n.translate('xpack.maps.metricSelect.minDropDownOptionLabel', {
      defaultMessage: 'Min',
    }),
    value: AGG_TYPE.MIN,
  },
  {
    label: i18n.translate('xpack.maps.metricSelect.percentileDropDownOptionLabel', {
      defaultMessage: 'Percentile',
    }),
    value: AGG_TYPE.PERCENTILE,
  },
  {
    label: i18n.translate('xpack.maps.metricSelect.sumDropDownOptionLabel', {
      defaultMessage: 'Sum',
    }),
    value: AGG_TYPE.SUM,
  },
  {
    label: i18n.translate('xpack.maps.metricSelect.termsDropDownOptionLabel', {
      defaultMessage: 'Top term',
    }),
    value: AGG_TYPE.TERMS,
  },
  {
    label: i18n.translate('xpack.maps.metricSelect.cardinalityDropDownOptionLabel', {
      defaultMessage: 'Unique count',
    }),
    value: AGG_TYPE.UNIQUE_COUNT,
  },
];

type Props = Omit<EuiComboBoxProps<AGG_TYPE>, 'onChange'> & {
  value: AGG_TYPE;
  onChange: (aggType: AGG_TYPE) => void;
  metricsFilter?: (metricOption: EuiComboBoxOptionOption<AGG_TYPE>) => boolean;
};

export function MetricSelect({ value, onChange, metricsFilter, ...rest }: Props) {
  function onAggChange(selectedOptions: Array<EuiComboBoxOptionOption<AGG_TYPE>>) {
    if (selectedOptions.length === 0) {
      return;
    }

    const aggType = selectedOptions[0].value!;
    onChange(aggType);
  }

  const options = metricsFilter ? AGG_OPTIONS.filter(metricsFilter) : AGG_OPTIONS;

  return (
    <EuiComboBox
      placeholder={i18n.translate('xpack.maps.metricSelect.selectAggregationPlaceholder', {
        defaultMessage: 'Select aggregation',
      })}
      singleSelection={true}
      isClearable={false}
      options={options}
      selectedOptions={AGG_OPTIONS.filter((option) => {
        return value === option.value;
      })}
      onChange={onAggChange}
      {...rest}
    />
  );
}
