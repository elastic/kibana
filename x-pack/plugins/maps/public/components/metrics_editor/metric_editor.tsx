/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import React, { ChangeEvent, Fragment } from 'react';
import { i18n } from '@kbn/i18n';

import { EuiButtonEmpty, EuiComboBoxOptionOption, EuiFieldText, EuiFormRow } from '@elastic/eui';

import { FormattedMessage } from '@kbn/i18n/react';
import { MetricSelect } from './metric_select';
import { SingleFieldSelect } from '../single_field_select';
import { AggDescriptor } from '../../../common/descriptor_types';
import { AGG_TYPE, DEFAULT_PERCENTILE } from '../../../common/constants';
import { getTermsFields } from '../../index_pattern_util';
import { IFieldType } from '../../../../../../src/plugins/data/public';
import { ValidatedNumberInput } from '../validated_number_input';

function filterFieldsForAgg(fields: IFieldType[], aggType: AGG_TYPE) {
  if (!fields) {
    return [];
  }

  if (aggType === AGG_TYPE.UNIQUE_COUNT || aggType === AGG_TYPE.TERMS) {
    return getTermsFields(fields);
  }

  const metricAggFieldTypes = ['number'];
  if (aggType !== AGG_TYPE.SUM) {
    metricAggFieldTypes.push('date');
  }

  return fields.filter((field) => {
    return field.aggregatable && metricAggFieldTypes.includes(field.type);
  });
}

interface Props {
  metric: AggDescriptor;
  fields: IFieldType[];
  onChange: (metric: AggDescriptor) => void;
  onRemove: () => void;
  metricsFilter?: (metricOption: EuiComboBoxOptionOption<AGG_TYPE>) => boolean;
  showRemoveButton: boolean;
}

export function MetricEditor({
  fields,
  metricsFilter,
  metric,
  onChange,
  showRemoveButton,
  onRemove,
}: Props) {
  const onAggChange = (metricAggregationType?: AGG_TYPE) => {
    if (!metricAggregationType) {
      return;
    }

    const descriptor = {
      type: metricAggregationType,
      label: metric.label,
    };

    if (metricAggregationType === AGG_TYPE.COUNT || !('field' in metric) || !metric.field) {
      onChange(descriptor);
      return;
    }

    const fieldsForNewAggType = filterFieldsForAgg(fields, metricAggregationType);
    const found = fieldsForNewAggType.find((field) => field.name === metric.field);
    const newDescriptor = {
      ...descriptor,
      field: found ? metric.field : undefined,
    };
    if (metricAggregationType === AGG_TYPE.PERCENTILE) {
      onChange({
        ...newDescriptor,
        percentile: 'percentile' in metric ? metric.percentile : DEFAULT_PERCENTILE,
      });
    } else {
      onChange(newDescriptor);
    }
  };
  const onFieldChange = (fieldName?: string) => {
    if (!fieldName || metric.type === AGG_TYPE.COUNT) {
      return;
    }
    onChange({
      label: metric.label,
      type: metric.type,
      field: fieldName,
    });
  };

  const onPercentileChange = (percentile: number) => {
    if (metric.type !== AGG_TYPE.PERCENTILE) {
      return;
    }
    onChange({
      ...metric,
      percentile,
    });
  };
  const onLabelChange = (e: ChangeEvent<HTMLInputElement>) => {
    onChange({
      ...metric,
      label: e.target.value,
    });
  };

  let fieldSelect;
  if (metric.type && metric.type !== AGG_TYPE.COUNT) {
    const showErrors = metric.field === undefined;
    fieldSelect = (
      <EuiFormRow
        label={i18n.translate('xpack.maps.metricsEditor.selectFieldLabel', {
          defaultMessage: 'Field',
        })}
        display="columnCompressed"
        isInvalid={showErrors}
        error={i18n.translate('xpack.maps.metricsEditor.selectFieldError', {
          defaultMessage: 'Field required for aggregation',
        })}
      >
        <SingleFieldSelect
          placeholder={i18n.translate('xpack.maps.metricsEditor.selectFieldPlaceholder', {
            defaultMessage: 'Select field',
          })}
          value={metric.field ? metric.field : null}
          onChange={onFieldChange}
          fields={filterFieldsForAgg(fields, metric.type)}
          isClearable={false}
          isInvalid={showErrors}
          compressed
        />
      </EuiFormRow>
    );
  }

  let percentileSelect;
  if (metric.type === AGG_TYPE.PERCENTILE) {
    const label = i18n.translate('xpack.maps.metricsEditor.selectPercentileLabel', {
      defaultMessage: 'Percentile',
    });
    percentileSelect = (
      <ValidatedNumberInput
        min={0}
        max={100}
        onChange={onPercentileChange}
        label={label}
        initialValue={
          typeof metric.percentile === 'number' ? metric.percentile : DEFAULT_PERCENTILE
        }
        display="columnCompressed"
      />
    );
  }

  let labelInput;
  if (metric.type) {
    labelInput = (
      <EuiFormRow
        label={i18n.translate('xpack.maps.metricsEditor.customLabel', {
          defaultMessage: 'Custom label',
        })}
        display="columnCompressed"
      >
        <EuiFieldText
          onChange={onLabelChange}
          value={metric.label ? metric.label : ''}
          compressed
        />
      </EuiFormRow>
    );
  }

  let removeButton;
  if (showRemoveButton) {
    removeButton = (
      <div className="mapMetricEditorPanel__metricRemoveButton">
        <EuiButtonEmpty
          iconType="trash"
          size="xs"
          color="danger"
          onClick={onRemove}
          aria-label={i18n.translate('xpack.maps.metricsEditor.deleteMetricAriaLabel', {
            defaultMessage: 'Delete metric',
          })}
        >
          <FormattedMessage
            id="xpack.maps.metricsEditor.deleteMetricButtonLabel"
            defaultMessage="Delete metric"
          />
        </EuiButtonEmpty>
      </div>
    );
  }

  return (
    <Fragment>
      <EuiFormRow
        label={i18n.translate('xpack.maps.metricsEditor.aggregationLabel', {
          defaultMessage: 'Aggregation',
        })}
        display="columnCompressed"
      >
        <MetricSelect
          onChange={onAggChange}
          value={metric.type}
          metricsFilter={metricsFilter}
          compressed
        />
      </EuiFormRow>

      {fieldSelect}
      {percentileSelect}
      {labelInput}
      {removeButton}
    </Fragment>
  );
}
