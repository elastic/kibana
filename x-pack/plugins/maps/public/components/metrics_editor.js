/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { i18n } from '@kbn/i18n';
import { FormattedMessage } from '@kbn/i18n/react';
import { EuiButtonEmpty, EuiSpacer, EuiTextAlign } from '@elastic/eui';
import { MetricEditor } from './metric_editor';
import { DEFAULT_METRIC } from '../classes/sources/es_agg_source';

export function MetricsEditor({ fields, metrics, onChange, allowMultipleMetrics, metricsFilter }) {
  function renderMetrics() {
    // There was a bug in 7.8 that initialized metrics to [].
    // This check is needed to handle any saved objects created before the bug was patched.
    const nonEmptyMetrics = metrics.length === 0 ? [DEFAULT_METRIC] : metrics;
    return nonEmptyMetrics.map((metric, index) => {
      const onMetricChange = (metric) => {
        onChange([...metrics.slice(0, index), metric, ...metrics.slice(index + 1)]);
      };

      const onRemove = () => {
        onChange([...metrics.slice(0, index), ...metrics.slice(index + 1)]);
      };

      let removeButton;
      if (index > 0) {
        removeButton = (
          <div className="mapMetricEditorPanel__metricRemoveButton">
            <EuiButtonEmpty
              iconType="trash"
              size="xs"
              color="danger"
              onClick={onRemove}
              aria-label={i18n.translate('xpack.maps.metricsEditor.deleteMetricAriaLabel', {
                defaultMessage: 'Delete metric',
              })}
            >
              <FormattedMessage
                id="xpack.maps.metricsEditor.deleteMetricButtonLabel"
                defaultMessage="Delete metric"
              />
            </EuiButtonEmpty>
          </div>
        );
      }
      return (
        <div key={index} className="mapMetricEditorPanel__metricEditor">
          <MetricEditor
            onChange={onMetricChange}
            metric={metric}
            fields={fields}
            metricsFilter={metricsFilter}
            removeButton={removeButton}
          />
        </div>
      );
    });
  }

  function addMetric() {
    onChange([...metrics, {}]);
  }

  function renderAddMetricButton() {
    if (!allowMultipleMetrics) {
      return null;
    }

    return (
      <>
        <EuiSpacer size="xs" />
        <EuiTextAlign textAlign="center">
          <EuiButtonEmpty onClick={addMetric} size="xs" iconType="plusInCircleFilled">
            <FormattedMessage
              id="xpack.maps.metricsEditor.addMetricButtonLabel"
              defaultMessage="Add metric"
            />
          </EuiButtonEmpty>
        </EuiTextAlign>
      </>
    );
  }

  return (
    <Fragment>
      <div className="mapMapLayerPanel__metrics">{renderMetrics()}</div>

      {renderAddMetricButton()}
    </Fragment>
  );
}

MetricsEditor.propTypes = {
  metrics: PropTypes.array,
  fields: PropTypes.array,
  onChange: PropTypes.func.isRequired,
  allowMultipleMetrics: PropTypes.bool,
  metricsFilter: PropTypes.func,
};

MetricsEditor.defaultProps = {
  metrics: [DEFAULT_METRIC],
  allowMultipleMetrics: true,
};
