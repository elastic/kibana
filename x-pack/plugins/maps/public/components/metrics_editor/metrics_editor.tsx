/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React, { Fragment } from 'react';
import { FormattedMessage } from '@kbn/i18n/react';
import { EuiButtonEmpty, EuiComboBoxOptionOption, EuiSpacer, EuiTextAlign } from '@elastic/eui';
import { MetricEditor } from './metric_editor';
import { DEFAULT_METRIC } from '../../classes/sources/es_agg_source';
import { IFieldType } from '../../../../../../src/plugins/data/public';
import { AggDescriptor } from '../../../common/descriptor_types';
import { AGG_TYPE } from '../../../common/constants';

interface Props {
  allowMultipleMetrics: boolean;
  metrics: AggDescriptor[];
  fields: IFieldType[];
  onChange: (metrics: AggDescriptor[]) => void;
  metricsFilter?: (metricOption: EuiComboBoxOptionOption<AGG_TYPE>) => boolean;
}

export function MetricsEditor({
  fields,
  metrics = [DEFAULT_METRIC],
  onChange,
  allowMultipleMetrics = true,
  metricsFilter,
}: Props) {
  function renderMetrics() {
    // There was a bug in 7.8 that initialized metrics to [].
    // This check is needed to handle any saved objects created before the bug was patched.
    const nonEmptyMetrics = metrics.length === 0 ? [DEFAULT_METRIC] : metrics;
    return nonEmptyMetrics.map((metric, index) => {
      const onMetricChange = (updatedMetric: AggDescriptor) => {
        onChange([...metrics.slice(0, index), updatedMetric, ...metrics.slice(index + 1)]);
      };

      const onRemove = () => {
        onChange([...metrics.slice(0, index), ...metrics.slice(index + 1)]);
      };

      return (
        <div key={index} className="mapMetricEditorPanel__metricEditor">
          <MetricEditor
            onChange={onMetricChange}
            metric={metric}
            fields={fields}
            metricsFilter={metricsFilter}
            showRemoveButton={index > 0}
            onRemove={onRemove}
          />
        </div>
      );
    });
  }

  function addMetric() {
    onChange([...metrics, { type: AGG_TYPE.AVG }]);
  }

  function renderAddMetricButton() {
    if (!allowMultipleMetrics) {
      return null;
    }

    return (
      <Fragment>
        <EuiSpacer size="xs" />
        <EuiTextAlign textAlign="center">
          <EuiButtonEmpty onClick={addMetric} size="xs" iconType="plusInCircleFilled">
            <FormattedMessage
              id="xpack.maps.metricsEditor.addMetricButtonLabel"
              defaultMessage="Add metric"
            />
          </EuiButtonEmpty>
        </EuiTextAlign>
      </Fragment>
    );
  }

  return (
    <Fragment>
      <div className="mapMapLayerPanel__metrics">{renderMetrics()}</div>

      {renderAddMetricButton()}
    </Fragment>
  );
}
