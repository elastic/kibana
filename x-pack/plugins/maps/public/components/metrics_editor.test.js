/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { MetricsEditor } from './metrics_editor';
import { AGG_TYPE } from '../../common/constants';

const defaultProps = {
  metrics: [
    {
      type: AGG_TYPE.SUM,
      field: 'myField',
    },
  ],
  fields: [],
  onChange: () => {},
  allowMultipleMetrics: true,
  metricsFilter: () => {},
};

test('should render metrics editor', async () => {
  const component = shallow(<MetricsEditor {...defaultProps} />);
  expect(component).toMatchSnapshot();
});

test('should add default count metric when metrics is empty array', async () => {
  const component = shallow(<MetricsEditor {...defaultProps} metrics={[]} />);
  expect(component).toMatchSnapshot();
});
