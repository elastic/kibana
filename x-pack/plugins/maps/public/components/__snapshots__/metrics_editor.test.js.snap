// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add default count metric when metrics is empty array 1`] = `
<Fragment>
  <div
    className="mapMapLayerPanel__metrics"
  >
    <div
      className="mapMetricEditorPanel__metricEditor"
      key="0"
    >
      <MetricEditor
        fields={Array []}
        metric={
          Object {
            "type": "count",
          }
        }
        metricsFilter={[Function]}
        onChange={[Function]}
      />
    </div>
  </div>
  <EuiSpacer
    size="xs"
  />
  <EuiTextAlign
    textAlign="center"
  >
    <EuiButtonEmpty
      iconType="plusInCircleFilled"
      onClick={[Function]}
      size="xs"
    >
      <FormattedMessage
        defaultMessage="Add metric"
        id="xpack.maps.metricsEditor.addMetricButtonLabel"
        values={Object {}}
      />
    </EuiButtonEmpty>
  </EuiTextAlign>
</Fragment>
`;

exports[`should render metrics editor 1`] = `
<Fragment>
  <div
    className="mapMapLayerPanel__metrics"
  >
    <div
      className="mapMetricEditorPanel__metricEditor"
      key="0"
    >
      <MetricEditor
        fields={Array []}
        metric={
          Object {
            "field": "myField",
            "type": "sum",
          }
        }
        metricsFilter={[Function]}
        onChange={[Function]}
      />
    </div>
  </div>
  <EuiSpacer
    size="xs"
  />
  <EuiTextAlign
    textAlign="center"
  >
    <EuiButtonEmpty
      iconType="plusInCircleFilled"
      onClick={[Function]}
      size="xs"
    >
      <FormattedMessage
        defaultMessage="Add metric"
        id="xpack.maps.metricsEditor.addMetricButtonLabel"
        values={Object {}}
      />
    </EuiButtonEmpty>
  </EuiTextAlign>
</Fragment>
`;
